ci: doc: restore tj-actions/changed-files to only build docs when doc files are changed

kartben opened the pull request:
Restore the use of tj-actions/changed-files action + related housekeeping.

str4t0m commented:
Thanks for coming up with a fix so quickly already, I was about to start crafting one up as well.
Looking the file history I noticed that this proposed approach was used before commit  a9f77a5 and was changed in order to be able to enforce the branch checks also when the job is skipped. Does that still apply?

kartben commented:
> Thanks for coming up with a fix so quickly already, I was about to start crafting one up as well. Looking the file history I noticed that this proposed approach was used before commit [a9f77a5](https://github.com/zephyrproject-rtos/zephyr/commit/a9f77a5126c5f7690eb76eeba5d4ca57842379cf) and was changed in order to be able to enforce the branch checks also when the job is skipped. Does that still apply?

good question - @nashif?

kartben commented:
mmh going back to draft - I think @str4t0m comment very much applies. If no "doc" file is touched then the workflow will not report an OK status causing PR to not be considered mergeable. Need to look more closely -- luckily we have a fix already deployed in main anyway

nashif commented:
probably we want to revert to the old state of things and use the same action as before, the way it was done solved many issues.

nashif commented:
> probably we want to revert to the old state of things and use the same action as before, the way it was done solved many issues.

ok, maybe I should have gone into the details, the problem here is that we want this check to pass, if the workflow is skipped because it does not match files, then GH will not allow PRs to be merged because it never got to running the workflow.

kartben committed:
Revert "ci: doc-build: do not check for changed files"

This reverts commit b80526658cc0e6ec5719f09cd8fff65d4e50525c
and restores the original use of looking for changed files
before triggering a documentation build.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
ci: doc: bump and pin tj-actions/changed-files action

Ensure we run the latest version of this action (and align with what is
already used for bsim-tests)

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: do not trigger doc-build on PR touching tests/

There is no need to trigger a doc build when tests are being
changed.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

pdgendt commented on .github/workflows/doc-build.yml:
```diff
@@ -21,10 +21,45 @@ env:
+      file_check: ${{ steps.check-doc-files.outputs.any_modified }}
+    steps:
+    - name: checkout
+      uses: actions/checkout@v4
```
Missing sha here.

kartben replied:
done

pdgendt commented on .github/workflows/doc-build.yml:
```diff
@@ -21,10 +21,45 @@ env:
+    needs: [doc-file-check]
     if: >
-      github.repository_owner == 'zephyrproject-rtos'
+      github.repository_owner == 'zephyrproject-rtos' &&
```
Should this check be repeated if it already needs `doc-file-check`?

kartben replied:
yeah actually I've removed the gating altogether -- workflow doesn't require a specific runner and being able to have doc built in ones personal fork can actually be quite convenient

kartben committed:
ci: doc: pin actions/checkout GH action

Following the revert in the previous commit, pin the actions/checkout
action to align with what is now the accepted practice in the project.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
ci: doc: do not prevent running doc CI in forks

Building docs just works on vanilla GH runner so it can be convenient
to not prevent the workflow to run on forks.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>
